plugins/catalog-react/src/types.ts

```diff
@@ -29,6 +30,13 @@ export type EntityFilter = {
     string | symbol | (string | symbol)[]
   >;
 
+  /**
+   * A superset of `getCatalogFilters` that supports filtering based on parameters outside of the
+   *  catalog filter system. Examples include `fullTextFilter` or `orderFields`.
+   * @returns
+   */
+  getBackendRequestParameter?: () => Record<string, JsonObject | JsonValue>;
```

Comment:
if we remove `getBackendRequestParameter` from here could makes things simpler (details in the other comments 😄)